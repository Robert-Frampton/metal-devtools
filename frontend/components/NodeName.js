import Component, {Config} from 'metal-jsx';

import otherProps from '../lib/otherProps';

export const OPENING = 0;
export const NORMAL_CLOSING = 1;
export const SELF_CLOSING = 2;

/**
 * Component used to display a component node
 */
class NodeName extends Component {
  render() {
    const {name, type} = this.props;

    return (
      <span class="node-name-container" {...otherProps(this)}>
        {type === NORMAL_CLOSING ? '</' : '<'}

        <span class="name">{name}</span>

        {type === SELF_CLOSING ? ' />' : '>'}
      </span>
    );
  }
}

NodeName.PROPS = {
  name: Config.string(),
  type: Config.oneOf([OPENING, NORMAL_CLOSING, SELF_CLOSING]).value(OPENING)
};

export default NodeName;
