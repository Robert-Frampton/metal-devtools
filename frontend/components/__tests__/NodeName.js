jest.unmock('../NodeName');

import NodeName, {OPENING, NORMAL_CLOSING, SELF_CLOSING} from '../NodeName';

describe('NodeName', () => {
  test('should render with OPENING', () => {
    const component = new NodeName({type: OPENING});

    expect(component).toMatchSnapshot();
  });

  test('should render with NORMAL_CLOSING', () => {
    const component = new NodeName({type: NORMAL_CLOSING});

    expect(component).toMatchSnapshot();
  });

  test('should render with SELF_CLOSING', () => {
    const component = new NodeName({type: SELF_CLOSING});

    expect(component).toMatchSnapshot();
  });
});
