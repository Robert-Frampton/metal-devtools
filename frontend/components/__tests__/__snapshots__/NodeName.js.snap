exports[`NodeName should render with NORMAL_CLOSING 1`] = `undefined`;

exports[`NodeName should render with OPENING 1`] = `undefined`;

exports[`NodeName should render with SELF_CLOSING 1`] = `undefined`;
